src/robusta/core/playbooks/prometheus_enrichment_utils.py

```diff
@@ -350,9 +351,38 @@ def create_chart_from_prometheus_query(
             dots_size=p.dots_size,
             stroke=p.stroke,
         )
-    return chart, PrometheusBlock(data=prometheus_query_result, query=promql_query, y_axis_type=values_format,
-                                  vertical_lines=vertical_lines, horizontal_lines=horizontal_lines,
-                                  graph_name=chart.title, metrics_legends_labels=metrics_legends_labels)
+    return chart, PrometheusBlock(
+        data=prometheus_query_result,
+        query=promql_query,
+        y_axis_type=values_format,
+        vertical_lines=vertical_lines,
+        horizontal_lines=horizontal_lines,
+        graph_name=chart.title,
+        metrics_legends_labels=metrics_legends_labels,
+    )
+
+
+def run_prometheus_query(prometheus_params: PrometheusParams, query: str) -> PrometheusQueryResult:
+    """
+    This function runs prometheus query and returns the result (usually a vector),
+    For graphs use run_prometheus_query_range which uses the prometheus query_range api
+    """
+    try:
+        prom = get_prometheus_connect(prometheus_params)
+        query = __add_additional_labels(query, prometheus_params)
+        prom_params = {"timeout": PROMETHEUS_REQUEST_TIMEOUT_SECONDS}
+        prom.check_prometheus_connection(prom_params)
+        results = prom.safe_custom_query(query=query, params=prom_params)
+        return PrometheusQueryResult(results)
+    except Exception as e:
```

Comment:
I think I would not surround with try/except
We have a global code to handle Prometheus errors

Also, someone calling the api might not expect having the error as a field of the result